Automattic/newspack-multibranded-site
# feat: add post primary brand UI

Asking for review from people with more Gutenberg experience to have a look at this.

Known issue: When you save the post, the SELECT temporarily switches its value to "none" and then goes back to the selected value.

I started this out as a modifier to the core's Taxonomy editor. you can see it on https://github.com/Automattic/newspack-multibranded-site/pull/12/commits/9d210835ee1ffc978a32f40ad495f22aa0f29f71

The problem was that Yoast also added a similar option, and then I had to treat it to a void a super confusing interface.

Since I was building the whole thing from scratch anyway, I figured a better solution would be to simply use this as the sidebar panel and disable the native one. (This was mostly copied from https://github.com/WordPress/gutenberg/blob/trunk/packages/editor/src/components/post-taxonomies/hierarchical-term-selector.js)

How to test:

* Create 2 brands
* Edit a post
* Confirm the Brands selector on the right works. (Select one brand, save the post, visit the post and confirm you see that brand)
* Confirm the "Manage Brands link" takes you to the right place
* Confirm that when you select 2 or more brands a "Primary brand" selector shows up
* Try saving the post with different options for Primary brand and confirm it works
* If "none" is selected, the post should be rendered with the default site layout
* If one brand is selected, this brand should be rendered in the post
* Activate Yoast and make sure it doesn't add a second "Primary brand" selector to the UI
